identity_resolver/src/resolution/domain_linkage_verifier.rs

```diff
@@ -0,0 +1,278 @@
+        }
+        // Resolve issuer
+        let issuer_resolved = self.resolve_issuer(&issuer_did).await?;
+        issuer_resolved
+          .verify_data(credential, &VerifierOptions::default())
+          .map_err(|err| {
+            Error::new(ErrorCause::DNSVerificationError {
+              reason: format!("Signature verification failed! {}", err),
+            })
+          })?;
+      }
+      Err(_e) => {
+        return Err(Error::new(ErrorCause::DNSVerificationError {
+          reason: "Invalid issuer".to_owned(),
+        }));
+      }
+    }
+    Ok(())
+  }
+
+  async fn resolve_issuer<D: DID>(&self, issuer_did: &D) -> Result<AbstractThreadSafeValidatorDocument> {
+    let resolved_doc: AbstractThreadSafeValidatorDocument = self.resolver.resolve(issuer_did).await.map_err(|err| {
+      Error::new(ErrorCause::DNSVerificationError {
+        reason: format!("Issuer resolution failed! {}", err),
+      })
+    })?;
+    Ok(resolved_doc)
+  }
+}
+
+mod tests {
+  use identity_core::convert::FromJson;
+  use crate::error::Result;
+  use crate::resolution::domain_linkage_verifier::DomainLinkageVerifier;
+  use crate::Resolver;
+  use identity_core::crypto::KeyPair;
+  use identity_core::crypto::ProofOptions;
+  use identity_core::crypto::KeyType;
+  use identity_credential::credential::Credential;
+  use identity_credential::credential::DIDConfigurationResource;
+  use identity_did::verification::MethodScope;
+  use identity_iota_core::IotaDID;
+  use identity_iota_core::IotaDocument;
+  use identity_iota_core::IotaVerificationMethod;
+
+  #[tokio::test]
+  pub async fn test() {
+    let configuration_string: &str = include_str!("./tests/fixtures/did_configuration/config1.json");
+    let mut configuration: DIDConfigurationResource =
+      DIDConfigurationResource::from_json(configuration_string).unwrap();
+
+    // let mut configuration = DomainLinkageVerifier::fetch_configuration(Url::parse("http://192.168.178.23:8080/".to_owned()).unwrap()).await.unwrap();
+    let credential: &mut Credential = configuration.linked_dids_mut().get_mut(0).unwrap();
+    let (doc, keypair) = compose_did_document(credential.issuer.url().to_string());
+
+    // Add valid proof from the created DID.
+    doc
+      .sign_data(credential, keypair.private(), "#key-1", ProofOptions::default())
+      .unwrap();
+
+    verify_config(doc, &configuration).await.unwrap();
+  }
+
+  #[tokio::test]
+  pub async fn no_proof() {
+    let configuration_string: &str = include_str!("./tests/fixtures/did_configuration/config1.json");
+    let mut configuration: DIDConfigurationResource =
+      DIDConfigurationResource::from_json(configuration_string).unwrap();
+    let credential: &mut Credential = configuration.linked_dids_mut().get_mut(0).unwrap();
+    let (doc, _) = compose_did_document(credential.issuer.url().to_string());
+    let res = verify_config(doc, &configuration).await;
+    assert!(res.is_err());
+  }
+
+  // Todo: Add further tests for error cases.
```

Comment:
Indeed it would be great to verify that the cases that `verify_domain_linkage_credential` seems to catch are indeed caught. You can setup a general function that does the generic setup (load credential, create issuer document, create configuration resource) and then take a closure that modifies a small part of the test and verify the result with another closure.